samples/vptr.cpp

```diff
@@ -29,92 +29,103 @@
-      cgh.parallel_for<init_b>(range<1>{N * M}, [=](item<1> index) {
-        accB[index] = index[0] * 2014;
+                                  access::target::global_buffer, float>(vptrB, cgh);
+      auto functor = [=](item<dim> index) {
+                       accB[index] = index[0] * multiplier2;
+                     };
+
+      cgh.parallel_for<init_b>(range<dim>{totalEntries}, functor);
       });
-    });
 
-    /* This kernel will perform the computation c = a + b. */
+    //This kernel will perform the computation c = a + b.
     myQueue.submit([&](handler& cgh) {
       auto accA = pMap.get_access<access::mode::read,
-                                  access::target::global_buffer, float>(a, cgh);
+                                  access::target::global_buffer, float>(vptrA, cgh);
       auto accB = pMap.get_access<access::mode::read,
-                                  access::target::global_buffer, float>(b, cgh);
+                                  access::target::global_buffer, float>(vptrB, cgh);
       auto accC = pMap.get_access<access::mode::discard_write,
-                                  access::target::global_buffer, float>(c, cgh);
-      cgh.parallel_for<matrix_add>(range<1>{N * M}, [=](item<1> index) {
-        accC[index] = accA[index] + accB[index];
-      });
+                                  access::target::global_buffer, float>(vptrC, cgh);
+
+      constexpr size_t dim = 1;
+      auto functor = [=](item<dim> index) {
+                       accC[index] = accA[index] + accB[index];
+                     };
+      
+      cgh.parallel_for<matrix_add>(range<dim>{totalEntries}, functor);
     });
 
-    /* On the host, the result stored in the buffer of virtual pointer "c" are
-     * checked. The matrix is accessed row by row, using pointer arithmetics on
-     * the virtual pointer. */
-    auto c_row = c;
-    for (size_t i = 0; i < N; i++) {
+    // On the host, the result stored in the buffer of virtual pointer "c" are
+    // checked. The matrix is accessed row by row, using pointer arithmetics on
+    // the virtual pointer.
+    auto c_row = vptrC;
+    for (size_t i = 0; i < row; ++i) {
       /* Get the number of elements by which the row is offset. */
       auto row_offset = pMap.get_element_offset<float>(c_row);
 
       /* Create a host accessor to access the data on the host. */
       auto accC = pMap.get_access<access::mode::read,
                                   access::target::host_buffer, float>(c_row);
-      for (size_t j = 0; j < M; j++) {
-        if (accC[row_offset + j] != (i * M + j) * (2 + 2014)) {
+      for (size_t j = 0; j < col; ++j) {
+        if (accC[row_offset + j] != (i * col + j) * (multiplier1 + multiplier2)) {
           std::cout << "Wrong value " << accC[row_offset + j] << " for element "
-                    << i * M + j << std::endl;
-          return -1;
+                    << i * col + j << '\n';
+          break;
```

Comment:
This needs to return an error code for ctest to check the correct execution of the sample.

Reply:
done :)